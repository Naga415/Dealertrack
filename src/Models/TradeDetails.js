'use strict';
export default class TradeDetails {
  constructor() {
    this.allowance = null;
    this.payoff = null;
    this.acv = null;
  }
}
