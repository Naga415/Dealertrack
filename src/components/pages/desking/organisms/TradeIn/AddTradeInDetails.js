'use strict';
import React from 'react';
import PropTypes from 'prop-types';

// eslint-disable-next-line react/prefer-stateless-function
class AddTradeInDetails extends React.PureComponent {
  constructor(props, context) {
    super(props, context);
  }
  
  render() {
    return (
      <div id={this.props.htmlId} className="add-tradein-details">
        <div>
          <div>A Trade In has not been provided for this deal</div>
        </div>
      </div>
    );
  }
}

AddTradeInDetails.propTypes = {
  htmlId: PropTypes.string.isRequired
};

export default AddTradeInDetails;
