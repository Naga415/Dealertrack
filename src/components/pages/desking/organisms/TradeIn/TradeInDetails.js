'use strict';
import React from 'react';
import PropTypes from 'prop-types';

// eslint-disable-next-line react/prefer-stateless-function
class TradeInDetails extends React.PureComponent {
  constructor(props, context) {
    super(props, context);
  }

  render() {
    return (
      <div id={this.props.htmlId}>
        TODO
      </div>
    );
  }
}

TradeInDetails.propTypes = {
  htmlId: PropTypes.string.isRequired
};

export default TradeInDetails;
