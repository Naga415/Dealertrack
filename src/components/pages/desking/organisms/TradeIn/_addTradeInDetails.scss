$addTradeInDetails:'.add-tradein-details'; // namespace for component

#{$addTradeInDetails} {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 240px;
}

