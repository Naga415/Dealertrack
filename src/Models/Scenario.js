'use strict';
import TradeDetails from './TradeDetails';

export default class Scenario {
  constructor() {
    this.id = null;
    this.type = null;
    this.summary = null;
    this.details = null;
    this.trade = new TradeDetails();
    this.frontProfitBreakdown = [];
    this.backProfitBreakdown = [];
  }
}
