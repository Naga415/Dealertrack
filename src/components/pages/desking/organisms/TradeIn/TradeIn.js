'use strict';
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Card from '@coxautokc/fusion-ui-components/lib/Card';
import Scenario from '../../../../../models/Scenario';
import AddTradeInDetails from './AddTradeInDetails';
import TradeInDetails from './TradeInDetails';

// eslint-disable-next-line react/prefer-stateless-function
export class TradeInClass extends React.PureComponent {
  constructor(props, context) {
    super(props, context);
  }

  render() {
    const showTradeIn = (this.props.scenario && this.props.scenario.trade);
    return (
      <Card
        htmlId="TradeInCard"
        header="TRADE IN">
        {showTradeIn?<TradeInDetails htmlId="TradeInDetails"/>:<AddTradeInDetails htmlId="AddTradeInDetails"/>}
      </Card>
    );
  }
}

function mapStateToProps(state) {
  return {
    scenario: state.deal.primaryScenario
  };
}

TradeInClass.propTypes = {
  htmlId: PropTypes.string,
  scenario: PropTypes.instanceOf(Scenario)
};

export default connect(
  mapStateToProps
)(TradeInClass);
