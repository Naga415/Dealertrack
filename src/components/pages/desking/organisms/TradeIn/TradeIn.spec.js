import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Scenario from '../../../../../models/Scenario';
import { TradeInClass as TradeIn } from './TradeIn';
import TradeDetails from '../../../../../models/TradeDetails';

describe('Trade In Card Tests', () => {
  const tradeDetails = new TradeDetails();
  tradeDetails.allowance = 20000;
  tradeDetails.payoff = 30000;
  tradeDetails.acv = 40000;

  const pScenario = new Scenario();
  pScenario.trade = tradeDetails;
  
  const htmlId = "TradeIn";
  const props = {
    htmlId,
    scenario: pScenario,
  };

  it('renders <TradeIn /> organism', () => {
    const component = renderer.create(<TradeIn {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('will render trade in content', () => {
    const component = mount(<TradeIn {...props} />);
    expect(component.find('#TradeInDetails').length).toBe(1);
  });

  it('will render add trade in content', () => {
    const props ={htmlId};
    const component = mount(<TradeIn {...props} />);
    expect(component.find('#AddTradeInDetails').length).toBe(1);
  });
});
