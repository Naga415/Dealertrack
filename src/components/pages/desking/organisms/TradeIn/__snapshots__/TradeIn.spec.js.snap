// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trade In Card Tests renders <TradeIn /> organism 1`] = `
<div
  className="card panel panel-default"
  id="TradeInCard"
>
  <div
    className="panel-heading"
  >
    <div
      className="card__title panel-title"
    >
      TRADE IN
    </div>
  </div>
  <div
    className="panel-body"
  >
    <div
      id="TradeInDetails"
    >
      TODO
    </div>
  </div>
</div>
`;
